cli/cmd/project/logs.go

```diff
@@ -0,0 +1,109 @@
+package project
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	"github.com/rilldata/rill/cli/pkg/config"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	runtimeclient "github.com/rilldata/rill/runtime/client"
+	"github.com/spf13/cobra"
+)
+
+func LogsCmd(cfg *config.Config) *cobra.Command {
+	var name, path string
+	var follow bool
+	var tail int
+
+	logsCmd := &cobra.Command{
+		Use:   "logs [<project-name>]",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Show project logs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := cmdutil.Client(cfg)
+			if err != nil {
+				return err
+			}
+			defer client.Close()
+
+			if len(args) > 0 {
+				name = args[0]
+			}
+
+			if !cmd.Flags().Changed("project") && len(args) == 0 && cfg.Interactive {
+				name, err = inferProjectName(cmd.Context(), client, cfg.Org, path)
+				if err != nil {
+					return err
+				}
+			}
+
+			proj, err := client.GetProject(context.Background(), &adminv1.GetProjectRequest{
+				OrganizationName: cfg.Org,
+				Name:             name,
+			})
+			if err != nil {
+				return err
+			}
+
+			depl := proj.ProdDeployment
+			if depl == nil {
+				return nil
+			}
+
+			if depl.Status != adminv1.DeploymentStatus_DEPLOYMENT_STATUS_OK {
+				cmdutil.PrintlnWarn(fmt.Sprintf("Deployment status not OK: %s", depl.Status.String()))
+				return nil
+			}
+
+			rt, err := runtimeclient.New(depl.RuntimeHost, proj.Jwt)
+			if err != nil {
+				return fmt.Errorf("failed to connect to runtime: %w", err)
+			}
+
+			if follow {
+				logClient, err := rt.WatchLogs(context.Background(), &runtimev1.WatchLogsRequest{InstanceId: depl.RuntimeInstanceId, Replay: true, ReplayLimit: int32(tail)})
+				if err != nil {
+					return fmt.Errorf("failed to watch logs: %w", err)
+				}
+
+				ctx, cancel := context.WithCancel(cmd.Context())
+
+				go func() {
+					for {
+						res, err := logClient.Recv()
+						if err != nil {
+							fmt.Println("failed to receive logs: %w", err)
+							cancel()
+							break
+						}
+						fmt.Println(res.Log)
+					}
+				}()
+
+				// keep on receiving logs util context is cancelled
+				<-ctx.Done()
+				return nil
+			}
+
+			res, err := rt.GetLogs(context.Background(), &runtimev1.GetLogsRequest{InstanceId: depl.RuntimeInstanceId, Ascending: true, Limit: int32(tail)})
+			if err != nil {
+				return fmt.Errorf("failed to get logs: %w", err)
+			}
+
+			for _, log := range res.Logs {
+				fmt.Println(log)
+			}
+			return nil
+		},
+	}
+
+	logsCmd.Flags().SortFlags = false
+	logsCmd.Flags().StringVar(&name, "project", "", "Project Name")
+	logsCmd.Flags().StringVar(&path, "path", ".", "Project directory")
+	logsCmd.Flags().BoolVarP(&follow, "follow", "f", false, "Follow logs")
+	logsCmd.Flags().IntVarP(&tail, "tail", "t", -1, "Number of lines to show from the end of the logs, use -1 for all logs")
```

Comment:
What about defaulting to something less, maybe 10 if `follow` is true and 50 if it's false?

Reply:
this is how kubectl logs work so just emulated it, but can make change if required

Reply:
Okay, maybe it's not so bad for the non-follow case. For follow, I think it would be a nicer experience to get quite few logs by default (since most of the time, you would use it to know what happens after running the command, not before). 

What do you think?

Reply:
no strong opinions here I am in habit of using kubectl logs, in any case lets keep the default value for tail consistent for follow and non-follow case